script.js

```diff
@@ -11,3 +45,471 @@ function hideColumnContents(className) {
+
+        for (const element of columnThree) {
+          element.style.width = columnThreeWidth + 'px';
+          headerThree.style.width = columnThreeWidth + 'px';
+        }
+      }
+
+      for (const element of columnOne) {
+        element.style.width = columnOneWidth + 'px';
+        headerOne.style.width = columnOneWidth + 'px';
+      }
+
+      for (const element of columnFour) {
+        element.style.width = columnFourWidth + 'px';
+        headerFour.style.width = columnFourWidth + 'px';
+      }
+    }
+  }
+
+  columnlineFour.onmousedown = function(event) {
+    var oldColumnlineFourPosition = columnlineFour.offsetLeft;
+    var columnOneWidth = headerOne.offsetWidth;
+    var columnTwoWidth = headerTwo.offsetWidth;
+    var columnThreeWidth = headerThree.offsetWidth;
+    var columnFourWidth = headerFour.offsetWidth;
+    var oldmousePosition = event.clientX;
+
+    function moveToMouse(windowCoordinatesX) {
+      const newColumnlineLeft = windowCoordinatesX;
+      const newColumnlineLeftIsInMinWidth = newColumnlineLeft >= headerThree.offsetLeft + minWidth && newColumnlineLeft <= headerFour.offsetLeft + headerFour.offsetWidth - minWidth;
+
+      if (newColumnlineLeftIsInMinWidth) {
+        columnlineFour.style.left = windowCoordinatesX  + 'px';
+      }
+      else {
+        document.removeEventListener('mousemove', onMouseMove);
+        const diffColumnlineFour = oldColumnlineFourPosition - columnlineFour.offsetLeft;
+
+        let diffIsNegative = diffColumnlineFour < 0;
+
+        if (diffIsNegative) {
+          columnThreeWidth = columnlineFour.offsetLeft - headerTwo.offsetWidth - headerOne.offsetWidth;
+          columnFourWidth = minWidth;
+          for (const element of columnThree) {
+            element.style.width = columnThreeWidth + 'px';
+          }
+
+          headerThree.style.width = columnThreeWidth + 'px';
+
+          for (const element of columnFour) {
+            element.style.width = columnFourWidth + 'px';
+            
+          }
+
+          headerFour.style.width = columnFourWidth + 'px';
+        }
+  
+        if (!diffIsNegative) {
+          columnThreeWidth = minWidth;
+          columnFourWidth = columnFourWidth + diffColumnlineFour;
+          for (const element of columnThree) {
+            element.style.width = columnThreeWidth + 'px';
+            
+          }
+
+          headerThree.style.width = columnThreeWidth + 'px';
+  
+          for (const element of columnFour) {
+            element.style.width = columnFourWidth + 'px';
+            
+          }
+
+          headerFour.style.width = columnFourWidth + 'px';
+        }
+        
+        for (const element of columnOne) {
+          element.style.width = columnOneWidth + 'px';
+        }
+
+        headerOne.style.width = columnOneWidth + 'px';
+  
+        for (const element of columnTwo) {
+          element.style.width = columnTwoWidth + 'px';
+        }
+
+        headerTwo.style.width = columnTwoWidth + 'px';
```

Comment:
same here. Repetitive code. use the a method. 

Do this for all appearances of repetitive code.